### DISCUSSION TITLE
Is it posible set max delay for restApi to get message from MQ?

### DISCUSSION THREAD

**Discussion by etkinator** [jmstoolbox/jmstoolbox#161] [Q&A]
“Is it possible to set the time for how long to wait for a message in MQ when reading a message from MQ? I would need to measure the time it takes for a response to arrive to a request sent via MQ.”

**Comment by titou10titou10** [comment]
> “Is it possible to set the time for how long to wait for a message in MQ when reading a message from MQ? 
What do you mean? I do not understand your need
JTB does not "wait" on messages, it "browses" the messages present on a queue. You can configure it to regularly "re-browse" a queues

**Comment by etkinator** [comment]
I need to get a message with a specific corrID via REST API and measure the time when it appears in MQ.

**Reply by titou10titou10** [reply]
what do you mean by "measure the time when it appears in MQ" ? 

Sorry I do not understand. You want to...
- measure how long it takes for your client app to post a message to MQ.? again, this has to be done on your client app. The message should be available to be retrieved by other apps as soon as your client app (producer) commit the workload  to MQ ?
- measure the time between the moment you send the REST request to JTB  and you receive the response (Consumer)? this has to be done on your client REST app..?

**Reply by etkinator** [reply]
I send mesaage to MQ and wiat fo answer from another aplication, that send answer to MQ with same corrID. I need measure time for this answer. I do it this way I reapat browse of MQ and wait for StatusCode 200 and must set maximum repat time for posibility that answer not deliver. I need to know if extist sipmliest way to do it.
